File: src/main/java/facts/stmt/JmpStmt.java
package facts.stmt;

import java.util.ArrayList;
import java.util.List;
import astnodes.exp.Expr;

/**
 * Jump
 */
public class JmpStmt extends Stmt {

    private String target;

    public JmpStmt(String pc, String target) {
        super(pc);
        this.target = target;
    }

    public String getTarget() {
        return target;
    }

    public void setTarget(String target) {
        this.target = target;
    }

    @Override
    public String toString() {
        return String.format("%sgoto label%s;", getLabel(), target);
    }

    @Override
    public List<Expr> getChildren() {
        return new ArrayList<>();
    }

    @Override
    public void replace(Expr oldExp, Expr newExp) {}
}

File: src/main/scala/astnodes/stmt/JmpStmt.scala
package astnodes.stmt

import astnodes.exp.Expr
import astnodes.exp.variable.Variable

/** Jump
  */
case class JmpStmt(override val pc: String, target: String) extends Stmt(pc) {
  override def toString: String = String.format("%sgoto label%s;", labelString, target)
  override def subst(v: Variable, w: Variable): Stmt = this
}
